Automattic/woocommerce-payments
# Add/232 partial capture api enhancement

Fixes https://github.com/Automattic/woocommerce-payments-meta/issues/232

#### Added 'capture amount' parameter to the capture authorization API endpoint

The existing capture authorization endpoint automatically captures the full order amount. However, there might be scenarios where a partial capture is desired. To address this, this PR introduces an optional `amount_to_capture` parameter. If omitted, the endpoint defaults to capturing the complete order amount. If provided, only the specified amount will be captured.

#### Testing instructions





- Ensure there are at least **two charges** available for capture.
- Using your preferred HTTP client, trigger the endpoint `wp-json/wc/v3/payments/orders/ORDER_ID/capture_authorization` with an uncaptured payment intent. Do **not** include any additional parameters like `amount_to_capture`. Confirm that the payment intent captures the entire order amount.
- Execute the same endpoint, but with a different order and payment intent. This time, include the `amount_to_capture` parameter and set it to a value of your choice. 
  - If the specified value exceeds the permissible capture amount, you will encounter an error.
  - Conversely, if the specified amount is less than the maximum capturable value, the payment intent should capture the designated amount.


-------------------


**Post merge**
